### ISSUE TITLE
Cannot set build feature from environment variable

### ISSUE THREAD

**Issue by AntoineD** [conda/conda-build#2400]
I tried to set a debug feature as explained [here](https://conda.io/docs/user-guide/tasks/build-packages/environment-variables.html#environment-variables-to-set-build-features):
`FEATURE_DEBUG=1 conda build xyz`
where the recipe of xyz has a another feature already set.
I get the following traceback:
```
Attempting to finalize metadata for xyz INFO:conda_build.metadata:Attempting to finalize metadata for xyz Traceback (most recent call last):
  File "/abc/anaconda2/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/abc/anaconda2/lib/python2.7/site-packages/conda_build/cli/main_build.py", line 388, in main
    execute(sys.argv[1:])
  File "/abc/anaconda2/lib/python2.7/site-packages/conda_build/cli/main_build.py", line 379, in execute
    noverify=args.no_verify)
  File "/abc/anaconda2/lib/python2.7/site-packages/conda_build/api.py", line 185, in build
    need_source_download=need_source_download, config=config, variants=variants)
  File "/abc/anaconda2/lib/python2.7/site-packages/conda_build/build.py", line 1771, in build_tree
    notest=notest,
  File "/abc/anaconda2/lib/python2.7/site-packages/conda_build/build.py", line 881, in build
    output_metas = expand_outputs([(m, need_source_download, need_reparse_in_env)])
  File "/abc/anaconda2/lib/python2.7/site-packages/conda_build/render.py", line 543, in expand_outputs
    for (output_dict, m) in _m.get_output_metadata_set(permit_unsatisfiable_variants=False):
  File "/abc/anaconda2/lib/python2.7/site-packages/conda_build/metadata.py", line 1757, in get_output_metadata_set
    permit_unsatisfiable_variants=permit_unsatisfiable_variants)
  File "/abc/anaconda2/lib/python2.7/site-packages/conda_build/metadata.py", line 638, in finalize_outputs_pass
    fm = finalize_metadata(om, permit_unsatisfiable_variants=permit_unsatisfiable_variants)
  File "/abc/anaconda2/lib/python2.7/site-packages/conda_build/render.py", line 289, in finalize_metadata
    permit_unsatisfiable_variants=permit_unsatisfiable_variants)
  File "/abc/anaconda2/lib/python2.7/site-packages/conda_build/render.py", line 112, in get_env_dependencies
    channel_urls=tuple(m.config.channel_urls))
  File "/abc/anaconda2/lib/python2.7/site-packages/conda_build/environ.py", line 658, in get_install_actions
    specs.append('%s@' % feature)
AttributeError: 'tuple' object has no attribute 'append'
```

**Comment by msarahan** [comment]
Sorry for the regression.  In general, using the new conda_build_config.yaml file will be better for controlling variants of packages.  Still, this is a bug we'll fix.  I'm not sure yet if we'll be expanding arbitrary variable support for environment variables.

**Comment by AntoineD** [comment]
Yes I already made the move to conda_build_config.yaml, it works fine.

**Comment by msarahan** [comment]
fixed by #2428, closing.
